cypress/integration/regressionTs/entityConflicts.spec.ts

```diff
@@ -241,41 +276,101 @@ describe('Entity Conflicts', () => {
 
                 // Create log dialogs (one for each test to isolate behavior)
                 (new Array<number>(4)).fill(0).forEach((_, i) => {
-                    logDialogs.CreateNewLogDialogButton()
-                    logDialog.TypeYourMessageValidateResponse(testData.userInput1, testData.actionResponse)
-                    // TODO: Why is this wait needed if we're waiting/validating response before next input
-                    cy.wait(500)
-                    logDialog.TypeYourMessageValidateResponse(testData.userInput2, testData.actionResponse)
-                    cy.wait(500)
-                    logDialog.TypeYourMessageValidateResponse(testData.userInput3, testData.actionResponse)
-                    logDialog.ClickDoneTestingButton()
+                    cy.get(s.logDialogs.buttonCreate)
+                        .click()
+
+                    cy.get(s.common.spinner, { timeout: constants.spinner.timeout })
+                        .should('not.exist')
+
+                    util.inputText(testData.userInput1)
+                    cy.get(s.webChat.messageFromBot, { timeout: constants.prediction.timeout })
+                        .contains(testData.actionResponse)
+
+                    util.inputText(testData.userInput2)
+                    cy.get(s.webChat.messageFromBot, { timeout: constants.prediction.timeout })
+                        .should($messages => {
+                            expect($messages).to.contain(testData.actionResponse)
+                            expect($messages).to.have.length(4)
+                        })
+
+                    util.inputText(testData.userInput3)
+                    cy.get(s.webChat.messageFromBot, { timeout: constants.prediction.timeout })
+                        .should($messages => {
+                            expect($messages).to.contain(testData.actionResponse)
+                            expect($messages).to.have.length(6)
+                        })
+
+                    cy.get(s.logDialog.buttonDone)
+                        .click()
+
+                    cy.get(s.logDialog.modal)
+                        .should('not.exist')
+
+                    cy.get(s.common.spinner, { timeout: constants.spinner.timeout })
+                        .should('not.exist')
                 })
 
                 // Change the labels on train dialogs to create conflict
-                model.NavigateToTrainDialogs();
+                cy.get(s.model.buttonNavTrainDialogs)
+                    .click()
 
                 cy.get(s.trainDialogs.descriptions)
                     .contains(testData.userInput1)
                     .click()
 
-                trainDialog.SelectChatTurnExactMatch(testData.userInput1)
-                trainDialog.RemoveEntityLabel(labeledWord1, testData.entityName)
-                trainDialog.LabelTextAsEntity(labeledWord2, testData.entityName)
-                trainDialog.ClickSubmitChangesButton()
```

Comment:
General comment for this code review.  If you decide to not employ the helpful functions, I'd add comments above each of the actions to explain what the action is doing.  i.e.  // Label Text As Entity

Reply:
Yea, in future I'll add comments for some of the multi step operations that are more obscure. 

> helpful functions

This assumes those functions are helpful.  They have nice names make the surface more appealing but going any deeper and you likely won't consider them helpful for any of the more valuable metrics like debuggability, composibility, documentation, etc.